feat: add `math/base/special/gammasgnf`

vivekmaurya001 opened the pull request:
Progresses #649 

## Description

- adds math/base/special/gammasgnf, which would be the single-precision equivalent for [math/base/special/gammasgn](https://github.com/stdlib-js/stdlib/tree/develop/lib/node_modules/%40stdlib/math/base/special/gammasgn).

This pull request:

- Progresses: https://github.com/stdlib-js/stdlib/issues/649

## Questions

> Any questions for reviewers of this pull request?

No.

## Other

> Any other information relevant to this pull request? This may include screenshots, references, and/or implementation notes.

No.

## Checklist

> Please ensure the following tasks are completed before submitting this pull request.

 - [x] Read, understood, and followed the [contributing guidelines](https://github.com/stdlib-js/stdlib/blob/develop/CONTRIBUTING.md)


* * *

@stdlib-js/reviewers 

[contributing]: https://github.com/stdlib-js/stdlib/blob/develop/CONTRIBUTING.md

vivekmaurya001 committed:
feat(math): add math/base/special/gammasgnf

vivekmaurya001 committed:
Merge branch 'gammasgnf' of https://github.com/vivekmaurya001/stdlib into gammasgnf

Planeshifter commented:
/stdlib update-copyright-years

stdlib-bot committed:
chore: update copyright years

vivekmaurya001 committed:
Merge branch 'stdlib-js:develop' into gammasgnf

vivekmaurya001 committed:
feat(add gammasgnf): add math/base/special/gammasgnf

vivekmaurya001 committed:
Merge branch 'gammasgnf' of https://github.com/vivekmaurya001/stdlib into gammasgnf

vivekmaurya001 committed:
Merge branch 'stdlib-js:develop' into gammasgnf

vivekmaurya001 committed:
add math/base/special/gammasgnf

---
type: pre_push_report
description: Results of running various checks prior to pushing changes.
report:
  - task: run_javascript_examples
    status: na
  - task: run_c_examples
    status: na
  - task: run_cpp_examples
    status: na
  - task: run_javascript_readme_examples
    status: na
  - task: run_c_benchmarks
    status: passed
  - task: run_cpp_benchmarks
    status: na
  - task: run_fortran_benchmarks
    status: na
  - task: run_javascript_benchmarks
    status: passed
  - task: run_julia_benchmarks
    status: na
  - task: run_python_benchmarks
    status: na
  - task: run_r_benchmarks
    status: na
  - task: run_javascript_tests
    status: passed
---

vivekmaurya001 commented:
/stdlib update-copyright-years

stdlib-bot committed:
chore: update copyright years

vivekmaurya001 committed:
add math/base/special/gammasgnf

vivekmaurya001 committed:
Merge branch 'gammasgnf' of https://github.com/vivekmaurya001/stdlib into gammasgnf

---
type: pre_push_report
description: Results of running various checks prior to pushing changes.
report:
  - task: run_javascript_examples
    status: na
  - task: run_c_examples
    status: na
  - task: run_cpp_examples
    status: na
  - task: run_javascript_readme_examples
    status: passed
  - task: run_c_benchmarks
    status: na
  - task: run_cpp_benchmarks
    status: na
  - task: run_fortran_benchmarks
    status: na
  - task: run_javascript_benchmarks
    status: na
  - task: run_julia_benchmarks
    status: na
  - task: run_python_benchmarks
    status: na
  - task: run_r_benchmarks
    status: na
  - task: run_javascript_tests
    status: na
---

vivekmaurya001 committed:
add math/base/special/gammasgnf

vivekmaurya001 commented:
Hi @gunjjoshi  , could u please help me from where error was occuring , i have double checked the addon.c file but not found anything

gunjjoshi commented on lib/node_modules/@stdlib/math/base/special/gammasgnf/src/addon.c:
```diff
@@ -0,0 +1,22 @@
+
+#include "stdlib/math/base/special/gammasgnf.h"
+#include "stdlib/math/base/napi/unary.h"
+
```
```suggestion

// cppcheck-suppress shadowFunction
```

gunjjoshi replied:
This should resolve the lint errors.

vivekmaurya001 replied:
thanks @gunjjoshi !! !

gunjjoshi committed:
Update lib/node_modules/@stdlib/math/base/special/gammasgnf/src/addon.c

Signed-off-by: Gunj Joshi <gunjjoshi8372@gmail.com>

vivekmaurya001 committed:
Merge branch 'stdlib-js:develop' into gammasgnf

vivekmaurya001 committed:
feat: add math/base/special/gammasgnf

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: passed
  - task: lint_package_json
    status: passed
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: na
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: passed
  - task: lint_javascript_tests
    status: na
  - task: lint_javascript_benchmarks
    status: passed
  - task: lint_python
    status: passed
  - task: lint_r
    status: na
  - task: lint_c_src
    status: missing_dependencies
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: missing_dependencies
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: na
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

vivekmaurya001 committed:
feat: add math/base/special/gammasgnf

vivekmaurya001 commented:
@gunjjoshi , I have maded changes according to my previous PR's reviews

anandkaranubc commented:
/stdlib merge

stdlib-bot committed:
Merge remote-tracking branch 'upstream/develop' into gammasgnf

anandkaranubc committed:
fix: code review

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: passed
  - task: lint_package_json
    status: passed
  - task: lint_repl_help
    status: passed
  - task: lint_javascript_src
    status: passed
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: passed
  - task: lint_javascript_tests
    status: passed
  - task: lint_javascript_benchmarks
    status: passed
  - task: lint_python
    status: passed
  - task: lint_r
    status: na
  - task: lint_c_src
    status: passed
  - task: lint_c_examples
    status: passed
  - task: lint_c_benchmarks
    status: passed
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: passed
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

kgryte commented on lib/node_modules/@stdlib/math/base/special/gammasgnf/lib/main.js:
```diff
@@ -0,0 +1,72 @@
+		return NaN;
+	}
+	if ( x > 0.0 ) {
+		return float64ToFloat32( 1.0 );
```
```suggestion
		return 1.0;
```

This wrapping is not necessary. `1.0` can be represented in single-precision.

kgryte commented on lib/node_modules/@stdlib/math/base/special/gammasgnf/lib/main.js:
```diff
@@ -0,0 +1,72 @@
+	}
+	fx = floorf( x );
+	if ( x === fx ) {
+		return float64ToFloat32( 0.0 );
```
```suggestion
		return 0.0;
```

kgryte commented on lib/node_modules/@stdlib/math/base/special/gammasgnf/lib/main.js:
```diff
@@ -0,0 +1,72 @@
+	}
+	fx = float64ToFloat32( fx / float64ToFloat32( 2.0 ) );
+	if ( fx === floorf( fx ) ) {
+		return float64ToFloat32( 1.0 );
```
```suggestion
		return 1.0;
```

kgryte commented on lib/node_modules/@stdlib/math/base/special/gammasgnf/lib/main.js:
```diff
@@ -0,0 +1,72 @@
+	if ( fx === floorf( fx ) ) {
+		return float64ToFloat32( 1.0 );
+	}
+	return float64ToFloat32( -1.0 );
```
```suggestion
	return -1.0;
```

kgryte commented on lib/node_modules/@stdlib/math/base/special/gammasgnf/lib/main.js:
```diff
@@ -0,0 +1,72 @@
+	if ( x === fx ) {
+		return float64ToFloat32( 0.0 );
+	}
+	fx = float64ToFloat32( fx / float64ToFloat32( 2.0 ) );
```
```suggestion
	fx = float64ToFloat32( fx / 2.0 );
```

`float64ToFloat32( 2.0 )` is a no-op.

kgryte committed:
Apply suggestions from code review

Signed-off-by: Athan <kgryte@gmail.com>

kgryte reviewed (approved):
LGTM
